tests/test_main.py

```diff
@@ -209,41 +256,64 @@ def test__get_expanded_columns_case_sensitive(self):
 
 class TestGetThreads(unittest.TestCase):
     def test__get_threads(self):
-        threaded, threads = _get_threads(None, None, None)
-        assert threaded
-        assert threads == 1
-
-        threaded, threads = _get_threads(None, 2, 3)
-        assert threaded
-        assert threads == 1
-
-        threaded, threads = _get_threads("serial", None, None)
-        assert not threaded
-        assert threads == 1
-
+        cli_options: CliOptions = get_cli_options()
+        cli_options.thread1 = None
+        cli_options.threads2 = None
+        _set_threads(cli_options)
+        assert cli_options.threaded
+        assert cli_options.threads == 1
+
+        cli_options.threads1 = 2
+        cli_options.threads2 = 3
+        _set_threads(cli_options)
+        assert cli_options.threaded
+        assert cli_options.threads == 1
+
+        cli_options.threads = "serial"
+        cli_options.threads1 = None
+        cli_options.threads2 = None
+        _set_threads(cli_options)
+        assert not cli_options.threaded
+        assert cli_options.threads == 1
+
+        cli_options.threads = "serial"
+        cli_options.threads1 = 1
+        cli_options.threads2 = 2
         with self.assertRaises(AssertionError):
-            _get_threads("serial", 1, 2)
-
-        threaded, threads = _get_threads("4", None, None)
-        assert threaded
-        assert threads == 4
+            _set_threads(cli_options)
 
+        cli_options.threads1 = None
+        cli_options.threads2 = None
+        cli_options.threads = "4"
         with self.assertRaises(ValueError) as value_error:
-            _get_threads("auto", None, None)
-        assert str(value_error.exception) == "invalid literal for int() with base 10: 'auto'"
-
-        threaded, threads = _get_threads(5, None, None)
-        assert threaded
-        assert threads == 5
-
-        threaded, threads = _get_threads(6, 7, 8)
-        assert threaded
-        assert threads == 6
+            _set_threads(cli_options)
+        assert str(value_error.exception) == "Error: threads must be of type int, or value must be 'serial'."
```

Comment:
pydantic will resolve this at runtime validation, but this custom error message is nice